ada_2_fhir-r4/fhir/2_fhir_fhir_include.xsl

```diff
@@ -54,6 +54,11 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
     </xd:doc>
     <xsl:param name="referencingStrategy" select="'uuid'" as="xs:string"/>
     
+    <xd:doc>
+        <xd:desc>Outputs all contained instances as separate files to filesystem.</xd:desc>
+    </xd:doc>
+    <xsl:param name="outputContained" select="false()" as="xs:boolean"/>
```

Comment:
Thing is, this doesn't belong on this level. Whether references are contained or not is defined on the source project level, and what you'd want to do with the result is defined by the target project. So this is the domain of the driver.

Which brings up another question; what do you want from the current drivers:
1. Interpret this ADA project and generate the corresponding FHIR instances (where the result can be one or more instances)?
2. Interpret this ADA project and save the resulting artifacts to disk?

At the moment, the saving to disk part is handled by a batch file (and by this commit). This could be generalized to something that takes the output and writes every resource to disk, using the id as the filename. Alternatively, this is added to the driver.

Anyway, I think for e.g. the CareTeam driver this commit can be replaced by adding something along the lines of
```xml
        <xsl:variable name="references" as="element()*">
            <xsl:for-each select="referenties/contactpersoon">
                <xsl:call-template name="nl-core-ContactPerson">
                    <xsl:with-param name="patient" select="$subject"/>
                </xsl:call-template>
            </xsl:for-each>
            <xsl:for-each select="referenties/patient">
                <xsl:call-template name="nl-core-Patient"/>
            </xsl:for-each>
            <xsl:for-each select="referenties/zorgverlener">
                <xsl:call-template name="nl-core-HealthProfessional-PractitionerRole"/>
                <xsl:call-template name="nl-core-HealthProfessional-Practitioner"/>
            </xsl:for-each>
            <xsl:for-each select="referenties/probleem">
                <xsl:call-template name="nl-core-Problem">
                    <xsl:with-param name="subject" select="$subject"/>
                </xsl:call-template>
            </xsl:for-each>
        </xsl:variable>
        <xsl:for-each select="$references">
            <xsl:result-document href="{./f:id/@value}.xml">
                <xsl:copy-of select="."/>
            </xsl:result-document>
        </xsl:for-each>
```

Reply:
Jup guess you're right.

I guess I put it here because I was looking for some generic mechanism that would work for all drivers. Because making and maintaining all of them becomes more and more of a hassle.

Will make an updated proposal, I guess I can do both after thinking about it